lucille-core/src/main/java/com/kmwllc/lucille/stage/JSoupExtractor.java

```diff
@@ -0,0 +1,103 @@
+package com.kmwllc.lucille.stage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.jsoup.Jsoup;
+import org.jsoup.select.Elements;
+import com.kmwllc.lucille.core.ConfigUtils;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.core.UpdateMode;
+import com.typesafe.config.Config;
+
+/**
+ * Selects elements from an HTML document and extracts the raw text from them into destination fields
+ * <br>
+ * Config Parameters -
+ * <br>
+ * <p>
+ * <b>filePathField</b> (String, Optional): field name which contains a path to an HTML document which will be processed
+ * </p>
+ * <p>
+ * <b>byteArrayField</b> (String, Optional): field name which contains a byte array of the html which will be processed (encoded in UTF-8).
+ * byteArrayField and filePathField cannot both be specified. If a document does not have whichever field is specified in the config, then it is 
+ * not modified
+ * </p>
+ * <b>destinationFields</b> (Map<String, String>) : defines a mapping from css selectors to the destination fields 
+ * in the processed document. If they already exist they are overwritten. Otherwise, they are created. If a selector returns 
+ * multiple elements the destination field receives a list of processed elements. If a selector returns no elements for a document then the 
+ * destination field is not created
+ * @see <a href="https://jsoup.org/cookbook/extracting-data/selector-syntax"> CSS selectors </a> for information on supported selectors
+ * </p>
+ */
+public class JSoupExtractor extends Stage {
```

Comment:
most stage names are phrased as verbs / actions 
ExtractHTML conforms to that approach, but you received a suggestion to include JSoup in the name because it is so well known
So... how about ApplyJSoup?